### PR TITLE
change event state from published to draft when date is cleared

### LINKED ISSUE #2264: Event state remains as scheduled when changing visibility to 'not visible or scheduled'
## Description

This was briefly mentioned in https://github.com/zetkin/app.zetkin.org/pull/2005#issue-2326323110

The Event state is not set correctly when setting and unsetting visibility date.

## Steps to reproduce
1. Go to https://app.dev.zetkin.org/organize/1/projects/
2. Click on any project
3. Create an event draft
5. Set visibility to a future date
6. Click outside the date picker
7. Open the date picker and press x to clear the date
8. Click outside the datepicker

### Expected Behaviour
Event state should change to default draft state

### Actual Behaviour
Event state remains as scheduled

## Screenshots (if you have any)

https://github.com/user-attachments/assets/652c9ca6-4580-4e3f-9fbf-c3e5532baad6

### PR DESCRIPTION
**Pull request by helihard**
## Description
This PR changes the event state from published to draft when user clears the date field in the date picker.

## Screenshots

https://github.com/user-attachments/assets/d78cce5b-05bd-4211-9c11-d7e64ae65c02

## Related issues
Resolves #2264

### CODE REVIEW

**Comment by helihard** [comment]
SORRY! Realised I might have created another bug, will give it another look when I get home.

**Comment by helihard** [comment]
Yes, I had actually created a new bug. That is hopefully fixed now.

(I noticed that if the user deleted the date in the input field – instead of clicking on the icon – and then clicks outside the date picker, the date remains unchanged. Deleting the date in that way would probably be by mistake anyway so I don't think it's a problem!)

**Comment by richardolsson** [comment]
Thanks for taking a closer look at this. I see from your commits that it's been challenging to get this to work properly.

I have tried the preview build that was generated from your first commit, and also the preview build that was generated from your second commit. I have only tried to reproduce the bug from the original bug report, and not looked into the bug that appeared with your fix.

When I try the build of your most recent commit, it does not seem to work. Clicking the X and then clicking outside the picker doesn't seem to do anything (just as described in the bug report).

That's why I went back and looked at the build of your previous commit as well, and that did fix the bug as far as I can tell (but I understand that you found it to introduce another bug).

Bottom line is that I don't think we're quite there yet. Please try to reproduce the bug on the most recent build and see if you agree.

**Comment by helihard** [comment]
Hm, I can reproduce the bug on main, but not on my branch. Clicking the X and then outside the picker changes the state from published to draft for me. I'm confused.

**Comment by helihard** [comment]
Happened upon another thing while trying it out: event is not unpublished on clearing the date (and therefore remains "Open" while simultaneously saying "Not visible or scheduled").
